apollo-router-core/src/services/http_compat.rs

```diff
@@ -118,15 +62,28 @@ impl<T> Request<T>
 where
     T: Default,
 {
-    // Only used for plugin_utils and tests
+    // Only used for plugin::utils and tests
     pub fn mock() -> Request<T> {
         Request {
-            url: Uri::from_str("http://default").unwrap(),
             inner: http::Request::default(),
         }
     }
 }
 
+impl<T> Deref for Request<T> {
```

Comment:
Get rid of these and restore the behaviour in compat Request? (Or at least restore the same behaviour)